packages/std/src/testing/mock.rs

```diff
@@ -244,6 +268,35 @@ impl Api for MockApi {
     }
 }
 
+impl MockApi {
+    /// Returns an address built from provided input string.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use cosmwasm_std::Addr;
+    /// # use cosmwasm_std::testing::MockApi;
+    /// #
+    /// let mock_api = MockApi::default();
+    /// let addr = mock_api.addr_make("creator");
+    ///
+    /// assert_eq!("cosmwasm1h34lmpywh4upnjdg90cjf4j70aee6z8qqfspugamjp42e4q28kqs8s7vcp", addr.to_string());
+    /// ```
+    pub fn addr_make(&self, input: &str) -> Addr {
+        let digest = Sha256::digest(input).to_vec();
+        if self.bech32_input_length > 0 && self.bech32_input_length <= digest.len() {
+            if let Ok(address) = encode(
+                self.bech32_prefix,
+                (&digest[..self.bech32_input_length]).to_base32(),
+                self.bech32_variant,
+            ) {
+                return Addr::unchecked(address);
+            }
+        }
+        Addr::unchecked(input)
```

Comment:
Do we need this case? I think everything can run through the hasher.

Reply:
When we stick to single variant and single byte length, then we can run everything through bech32::encode and panic when this function fails, with message like "Generating address failed with reason {}". This would be aligned with the comment below.

Reply:
Even with a flexible byte size you can implement it in a way that you always use the hash. Since you can assert the `length > 0 && length <= 32`, you can just use a sub-slice. But yeah, I don't see why we need the flexibility.

Reply:
Ok, we stick to 32B. Provided simplified implementation.